### ISSUE TITLE
Miscellaneous problems with `agama config set`

### ISSUE THREAD

**Issue by mvidner** [openSUSE/agama#688]
I found these while trying to test #666.

Good behavior for starters:
```console
# journalctl -f &
# agama config set software.product=foo
Unknown product 'foo'. Available products: '["ALP-Dolomite", "Tumbleweed", "Leap16"]'
# agama config set software.baseProduct=foo
Unknown attribute 'software.base_product'
# agama config set install.product=foo
Unknown attribute 'install.product'
```

CLI syntax errors are silently ignored: 

```console
# agama config set storage magic  # no equal sign
# agama config set storage=magic  # no section-dot-value
```

Now observe the `config show` output. I would expect `config set` and `config add` to use the same vocabulary:

```console
# agama -f yaml config show
user:
  fullName: ''
  userName: ''
  password: ''
  autologin: false
root:
  sshPublicKey: null
software:
  product: ''
storage:
  lvm: null
  encryptionPassword: null
  devices: []
network:
  connections:
  - id: foo
    method: disabled
  - id: lo
    method: manual
    addresses:
    - 127.0.0.1/8
```

Users are bad: 
```console
# agama config set users.blah=foo   # OK
Unknown attribute 'users.blah'
# agama config set user.blah=foo    # WTF, no error
# agama config set users.password=foo  # fair enough
Unknown attribute 'users.password'
# agama config set user.password=foo   # silently fails, bad
```

Storage... seems to work in the config layer but fails somewhere deeper. Might be some outdated dependencies in my container

```console
# agama config set storage.LVM=true
I, [2023-08-03T11:18:10.525118 #16216]  INFO -- storage: Calculating storage proposal from D-Bus settings: {"LVM"=>true, "CandidateDevices"=>[], "EncryptionPassword"=>""}
D-Bus service error

Caused by:
    org.freedesktop.DBus.Error.Failed: undefined method `software_raids' for nil:NilClass; caused by 1 sender=:1.28 -> dest=org.opensuse.Agama.Storage1 serial=7 reply_serial= path=/org/opensuse/Agama/Storage1; interface=org.opensuse.Agama.Storage1.Proposal.Calculator; member=Calculate error_name=
# agama config set storage.lvm=true
I, [2023-08-03T11:18:30.624296 #16216]  INFO -- storage: Calculating storage proposal from D-Bus settings: {"EncryptionPassword"=>"", "CandidateDevices"=>[], "LVM"=>true}
D-Bus service error

Caused by:
    org.freedesktop.DBus.Error.Failed: undefined method `software_raids' for nil:NilClass; caused by 1 sender=:1.30 -> dest=org.opensuse.Agama.Storage1 serial=7 reply_serial= path=/org/opensuse/Agama/Storage1; interface=org.opensuse.Agama.Storage1.Proposal.Calculator; member=Calculate error_name=
# agama config set storage.lvm=false
I, [2023-08-03T11:18:37.469158 #16216]  INFO -- storage: Calculating storage proposal from D-Bus settings: {"CandidateDevices"=>[], "LVM"=>false, "EncryptionPassword"=>""}
D-Bus service error

Caused by:
    org.freedesktop.DBus.Error.Failed: undefined method `software_raids' for nil:NilClass; caused by 1 sender=:1.32 -> dest=org.opensuse.Agama.Storage1 serial=7 reply_serial= path=/org/opensuse/Agama/Storage1; interface=org.opensuse.Agama.Storage1.Proposal.Calculator; member=Calculate error_name=
```

Network: does not complain about a nonsense attribute

```console
# agama config set network.magic=true
e7e86cd938d4:/checkout # 11:19:49 [ERROR] Could not add/update the connection foo: D-Bus service error
11:19:49 [WARN] Ignoring network device 'lo' (unsupported type '32')
11:19:49 [WARN] Ignoring network device 'tap0' (unsupported type '16')
11:19:49 [INFO] Publishing network connection 'lo'
11:19:49 [INFO] Publishing network connection 'foo'
Aug 03 11:19:49 e7e86cd938d4 NetworkManager[6259]: <info>  [1691061589.4849] audit: op="connection-update" uuid="1b3bc8cd-dab9-40aa-ad72-4add2be7c5a2" name="foo" pid=16197 uid=0 result="success"
Aug 03 11:19:49 e7e86cd938d4 NetworkManager[6259]: <info>  [1691061589.4874] audit: op="connection-activate" uuid="1b3bc8cd-dab9-40aa-ad72-4add2be7c5a2" name="foo" result="fail" reason="No suitable device found for this connection (device lo not available because profile is not compatible with device (connection type is not "loopback"))."
```

**Comment by mvidner** [comment]
I then tried to fix the case of the missing equal sign, `agama config set storage.lvm true`

I identified one place where `key_to_scope` returns a Result such as `Err(... InvalidKeyName("true"))` but the `filter_map...ok()` discards the errors:
https://github.com/openSUSE/agama/blob/9c7a6ad7810e8d2b95dfb45ebbaae45515093c9d/rust/agama-cli/src/config.rs#L41

Several attempts to report them via `anyhow::Result<()>` failed to compile :sob:

**Comment by imobachgs** [comment]
Thanks for such a detailed report! Those issues are fixed in #689. About storage settings, we have a PBI to work on them.
